src/systems/filecoin_blockchain/storage_power_consensus/_index.md

```diff
@@ -44,54 +46,126 @@ More precisely,
 - Nominal power = claimed power, unless the miner is in DetectedFault or Challenged state. Nominal power is used to determine total network storage power for purposes of consensus minimum.
 - Consensus power = nominal power, unless the miner fails to meet consensus minimum, or is undercollateralized.
 
-{{<label tickets>}}
-## Tickets
-
-Tickets are used across the Filecoin protocol as sources of randomness:
+{{<label beacon_entries>}}
+## Beacon Entries
+
+The Filecoin protocol uses randomness produced by a {{<sref drand>}} beacon to seed unbiasable randomness seeds for use in the chain (see {{<sref randomness>}}).
+
+In turn these random seeds are used by:
+
+- The {{<sref sector_sealer>}} as SealSeeds to bind sector commitments to a given subchain.
+- The {{<sref post_generator>}} as PoStChallenges to prove sectors remain committed as of a given block.
+- The Storage Power subsystem as randomness in {{<sref leader_election>}} to determine their eligibility to mine a block.
+
+This randomness may be drawn from various Filecoin chain epochs by the respective protocols that use them according to their security requirements.
+
+It is important to note that a given Filecoin network and a given drand network need not have the same round time, i.e. blocks may be generated faster or slower by Filecoin than randomness is generated by drand.
+For instance, if the drand beacon is producing randomness twice as fast as Filecoin produces blocks, we might expect two random values to be produced in a Filecoin epoch, conversely if the Filecoin network is twice as fast as drand, we might expect a random value every other Filecoin epoch. Accordingly, depending on both networks' configurations, certain Filecoin blocks could contain multiple or no drand entries.
+
+In all cases, Filecoin blocks must include all drand beacon outputs generated since the last epoch in the `BeaconEntries` field of the block header. Any use of randomness from a given Filecoin epoch should use the last valid drand entry included in a Filecoin block, as follows:
+
+```go
+// Note that the below function is a temporary solution and likely to be
+// updated in future. 
+GetBeaconEntryForEpoch(epoch) []BeaconEntry {
+    var rand Randomness
+    for rand == nil {
+        seeds := GetBeaconEntriesForEpoch(epoch)
+        if seeds == nil {
+            epoch -= 1
+        } else {
+            rand = seeds[len(seeds)-1].Randomness
+        }
+    }
+    return rand
+}
+
+GetBeaconEntriesForEpoch(epoch) []BeaconEntry {
+    maxDrandRound := MaxBeaconRoundForEpoch(epoch)
+
+    if epoch == 0 {
+        // special case genesis
+        entries := []
+        rand := drand.Public(curr)
+        return append(entries, rand)
+    }
```

Comment:
3. (medium) The spec has an incorrect definition of the randomness (`curr` is undefined) used at entry 0, in addition, it doesn't seem like that there should be an entry at the genesis block.
  - TODO: check how this is implemented in lotus, if this is incorrect change it.

Reply:
Check out https://github.com/filecoin-project/lotus/blob/master/chain/beacon/beacon.go#L54.

Updated spec given error you found. Nice catch. Should match lotus logic (checking on one uncertainty).